Similarities checker with "ignore-signatures" option enabled ignores functions with docstring-only bodies

#4648 follow-up.

### Steps to reproduce

Given multiple files:
```
module\
    __init__.py
    a.py
    b.py
```

`a.py`:
```python
def example_func(
    arg1,
    arg2,
    arg3,
    arg4,
):
    """Valid function definition with docstring only."""
```

`b.py`:
```python
def example_func(
    arg1,
    arg2,
    arg3,
    arg4,
):
    """Valid function definition with docstring only."""
```

### Current behavior

Result of `pylint --disable=all --enable=similarities --ignore-signatures=yes module`:
```
************* Module module.b
module/b.py:1:0: R0801: Similar lines in 2 files
==module.a:0
==module.b:0
def example_func(
    arg1,
    arg2,
    arg3,
    arg4,
):
    """Valid function definition with docstring only.""" (duplicate-code)

-----------------------------------
Your code has been rated at 5.00/10
```

### Expected behavior

`duplicate-code` error shouldn't be reported in this case.

**Note**: if functions have bodies everything works correctly. You could add `pass` to the example functions above and no `duplicate-code` error would be reported.

### pylint --version output

Result of `pylint --version` output:

```
pylint 2.9.3
astroid 2.6.2
Python 3.8.2 (default, Jun  8 2021, 11:59:35) 
[Clang 12.0.5 (clang-1205.0.22.11)]
```

--- a/CONTRIBUTORS.txt
+++ b/CONTRIBUTORS.txt
@@ -505,3 +505,6 @@ contributors:
 * Lorena Buciu (lorena-b): contributor
 
 * Sergei Lebedev (superbobry): contributor
+
+* Maksym Humetskyi (mhumetskyi): contributor
+  - Fixed ignored empty functions by similarities checker with "ignore-signatures" option enabled

--- a/ChangeLog
+++ b/ChangeLog
@@ -27,6 +27,10 @@ Release date: TBA
 
   Closes #4657
 
+* Fix ignored empty functions by similarities checker with "ignore-signatures" option enabled
+
+  Closes #4652
+
 
 What's New in Pylint 2.9.3?
 ===========================

--- a/pylint/checkers/similar.py
+++ b/pylint/checkers/similar.py
@@ -223,9 +223,11 @@ def stripped_lines(
         signature_lines = set(
             chain(
                 *(
-                    range(func.fromlineno, func.body[0].lineno)
+                    range(
+                        func.fromlineno,
+                        func.body[0].lineno if func.body else func.tolineno + 1,
+                    )
                     for func in functions
-                    if func.body
                 )
             )
         )

--- a/tests/checkers/unittest_similar.py
+++ b/tests/checkers/unittest_similar.py
@@ -35,6 +35,8 @@ SIMILAR3 = str(INPUT / "similar3")
 SIMILAR4 = str(INPUT / "similar4")
 SIMILAR5 = str(INPUT / "similar5")
 SIMILAR6 = str(INPUT / "similar6")
+EMPTY_FUNCTION_1 = str(INPUT / "similar_empty_func_1.py")
+EMPTY_FUNCTION_2 = str(INPUT / "similar_empty_func_2.py")
 MULTILINE = str(INPUT / "multiline-import")
 HIDE_CODE_WITH_IMPORTS = str(INPUT / "hide_code_with_imports.py")
 
@@ -195,6 +197,44 @@ def test_ignore_signatures_pass():
         output.getvalue().strip()
         == """
 TOTAL lines=29 duplicates=0 percent=0.00
+""".strip()
+    )
+
+
+def test_ignore_signatures_empty_functions_fail():
+    output = StringIO()
+    with redirect_stdout(output), pytest.raises(SystemExit) as ex:
+        similar.Run([EMPTY_FUNCTION_1, EMPTY_FUNCTION_2])
+    assert ex.value.code == 0
+    assert (
+        output.getvalue().strip()
+        == (
+            '''
+6 similar lines in 2 files
+==%s:1
+==%s:1
+       arg1: int = 1,
+       arg2: str = "2",
+       arg3: int = 3,
+       arg4: bool = True,
+   ) -> None:
+       """Valid function definition with docstring only."""
+TOTAL lines=14 duplicates=6 percent=42.86
+'''
+            % (EMPTY_FUNCTION_1, EMPTY_FUNCTION_2)
+        ).strip()
+    )
+
+
+def test_ignore_signatures_empty_functions_pass():
+    output = StringIO()
+    with redirect_stdout(output), pytest.raises(SystemExit) as ex:
+        similar.Run(["--ignore-signatures", EMPTY_FUNCTION_1, EMPTY_FUNCTION_2])
+    assert ex.value.code == 0
+    assert (
+        output.getvalue().strip()
+        == """
+TOTAL lines=14 duplicates=0 percent=0.00
 """.strip()
     )
 

--- a/tests/input/similar_empty_func_1.py
+++ b/tests/input/similar_empty_func_1.py
@@ -1,0 +1,7 @@
+def func1(
+    arg1: int = 1,
+    arg2: str = "2",
+    arg3: int = 3,
+    arg4: bool = True,
+) -> None:
+    """Valid function definition with docstring only."""

--- a/tests/input/similar_empty_func_2.py
+++ b/tests/input/similar_empty_func_2.py
@@ -1,0 +1,7 @@
+def func2(
+    arg1: int = 1,
+    arg2: str = "2",
+    arg3: int = 3,
+    arg4: bool = True,
+) -> None:
+    """Valid function definition with docstring only."""

